### PR TITLE
Validate application package before direct deployment

### PR DESCRIPTION
**Pull request by mpolden**
@jonmv

### CODE REVIEW

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/ControllerTest.java [hunk 1]
```diff
@@ -696,7 +696,7 @@ void testDnsUpdatesForApplicationEndpoint() {
 
     @Test
     void testDevDeployment() {
-        ApplicationPackage applicationPackage = new ApplicationPackageBuilder().build();
+        ApplicationPackage applicationPackage = ApplicationPackageBuilder.fromDeploymentXml(DeploymentSpec.empty.xmlForm());
```
Comment (jonmv) [inline controller-server/src/test/java/com/yahoo/vespa/hosted/controller/ControllerTest.java:699]: How about no `deployment.xml` at all?

**Review by jonmv — COMMENTED** [review]
One question, otherwise good :)

File: controller-server/src/test/java/com/yahoo/vespa/hosted/controller/ControllerTest.java (hunk 1 above)
Comment (mpolden) [reply controller-server/src/test/java/com/yahoo/vespa/hosted/controller/ControllerTest.java]: It's fine, but I updated the test to use a completely empty package.

**Comment by jonmv** [comment]
Great!

**Pull request merged**